### DISCUSSION TITLE
Specifying the download location ...

### DISCUSSION THREAD

**Discussion by thisguyshouldworkforus** [JuanBindez/pytubefix#39] [Q&A] [answered]
How do I specify the target download directory for the Audio/Video objects?

Despite the code below, I cannot seem to change either the download filename
  - desired name would be `yt.publish_date`.audio.webm // `yt.publish_date`.video.webm

Or the target download directory
  - desired path would be `/opt/projects/mytube/downloads` as I tried to input to the helper function, but that doesn't seem to do anything.

```python
    pytubefix.helpers.target_directory('/opt/projects/mytube/downloads')
    
    yt = pytubefix.YouTube(str(VIDEO), use_oauth=True, allow_oauth_cache=True)
    
    # Construct the Audio Object
    yta = yt.streams.filter(adaptive=True, mime_type="audio/webm", abr="160kbps").first().download()
    
    # Construct the Video Object
    ytv = yt.streams.filter(adaptive=True, mime_type="video/webm",res="1080p",).first().download()
```

Any / all help on this topic would be greatly appreciated!
Thank you.

**Comment by JuanBindez** [comment, marked as answer]
> How do I specify the target download directory for the Audio/Video objects?
> 
> Despite the code below, I cannot seem to change either the download filename
> 
>     * desired name would be `yt.publish_date`.audio.webm // `yt.publish_date`.video.webm
> 
> 
> Or the target download directory
> 
>     * desired path would be `/opt/projects/mytube/downloads` as I tried to input to the helper function, but that doesn't seem to do anything.
> 
> 
> ```python
>     pytubefix.helpers.target_directory('/opt/projects/mytube/downloads')
>     
>     yt = pytubefix.YouTube(str(VIDEO), use_oauth=True, allow_oauth_cache=True)
>     
>     # Construct the Audio Object
>     yta = yt.streams.filter(adaptive=True, mime_type="audio/webm", abr="160kbps").first().download()
>     
>     # Construct the Video Object
>     ytv = yt.streams.filter(adaptive=True, mime_type="video/webm",res="1080p",).first().download()
> ```
> 
> Any / all help on this topic would be greatly appreciated! Thank you.

try:     

```python
    from pytubefix import YouTube
    
    file_path = '/opt/projects/mytube/downloads'
    VIDEO = "url here"
        
    yt = YouTube(VIDEO, use_oauth=True, allow_oauth_cache=True)
        
    # Construct the Audio Object
    yta = yt.streams.filter(adaptive=True, mime_type="audio/webm"abr="160kbps").first().download(output_path=file_path)
        
    # Construct the Video Object
    ytv = yt.streams.filter(adaptive=True, mime_type="video/  webm",res="1080p",).first().download(output_path=file_path)
```

**Reply by thisguyshouldworkforus** [reply]
Yes, it wasn't until after I asked the question that I looked at the `streams.py` file itself. For whatever reason, VSCode never popped up the hint box to show the different parameters, so I was lost. But I quickly learned it was the `output_path` and `filename` that I failed to set.

So simple.  Thank you for your great fork!
